doc/changelog.md

```diff
@@ -137,75 +227,81 @@
-  [(#479)](https://github.com/PennyLaneAI/catalyst/pull/479)
+  - `compile_from_mlir(ir[, compiler, ...])` to compile a Catalyst function to binary code from the
+    provided MLIR.
 
-* Add support for post-selection in mid-circuit measurements. This is currently supported only on Lightning simulators.
-  [(#491)](https://github.com/PennyLaneAI/catalyst/pull/491)
+  - `filter_static_args(args, static_argnums)` to remove static values from arguments using the
+    provided index list.
 
-  This is an example of post-selection usage:
+  - `get_cmain(fn, *args)` to return a C program that calls a jitted function with the provided
+    arguments.
 
-  ```py
-  import pennylane as qml
-  from catalyst import qjit
+  - `print_compilation_stage(fn, stage)` to print one of the recorded compilation stages for a
+    JIT-compiled function.
 
-  dev = qml.device("lightning.qubit", wires=1)
+  - `promote_arguments(target_signature, args)` to promote arguments to the provided target
+    signature, preserving PyTrees.
 
-  @qjit
-  @qml.qnode(dev)
-  def f():
-      qml.Hadamard(0)
-      m = measure(0, postselect=1)
-      return qml.expval(qml.PauliZ(0))
-  ```
+  For more details, please see the `catalyst.debug` documentation.
 
-* Add support for reset option in mid-circuit measurements.
-  [(#507)](https://github.com/PennyLaneAI/catalyst/pull/507)
+* Remove copies of TOML device configuration files for Lightning device in Catalyst.
+  [(#472)](https://github.com/PennyLaneAI/catalyst/pull/472)
 
-  This is an example of reset usage:
+* Capturing quantum circuits with many gates prior to compilation is now quadratically faster (up to
+  a factor), by removing `qextract_p` and `qinst_p` from forced-order primitives.
+  [(#469)](https://github.com/PennyLaneAI/catalyst/pull/469)
 
-  ```py
-  dev = qml.device("lightning.qubit", wires=1)
+* Update `AllocateQubit` and `AllocateQubits` in `LightningKokkosSimulator` to preserve
+  the current state-vector before qubit re-allocations.
+  [(#479)](https://github.com/PennyLaneAI/catalyst/pull/479)
 
-  @qjit
-  @qml.qnode(dev)
-  def f():
-      qml.Hadamard(0)
-      m = measure(0, reset=True)
-      return qml.expval(qml.PauliZ(0))
-  ```
+* The [PennyLane custom compiler entry point name convention has changed](https://github.com/PennyLaneAI/pennylane/pull/5140), necessitating
+  a change to the Catalyst entry points.
+  [(#493)](https://github.com/PennyLaneAI/catalyst/pull/493)
 
 <h3>Breaking changes</h3>
 
+* Catalyst gradient functions now match the Jax convention for the returned axes of
+  gradients, Jacobians, VJPs, and JVPs. As a result, the returned tensor shape from various
+  Catalyst gradient functions may differ compared to previous versions of Catalyst.
+  [(#500)](https://github.com/PennyLaneAI/catalyst/pull/500)
+  [(#501)](https://github.com/PennyLaneAI/catalyst/pull/501)
+  [(#508)](https://github.com/PennyLaneAI/catalyst/pull/508)
+
+<h3>Internal changes</h3>
```

Comment:
@dime10 @maliasadi @erick-xanadu I added a new section to the changelog here, 'Internal Changes', because typically the Breaking Changes section refers to things that break in the 'public API' (e.g., user-facing Python functions, and device plugin API).

Whereas a lot of these entries seem more implementation details/internal changes with no implications on the public API

Reply:
Seems like a good idea, but I think one those things could still be considered breaking the public API, which is moving the debug methods on the QJIT class.

Reply:
But we also have other "public" interfaces to consider no?
- the QuantumDevice interface in the runtime for device plugins
- the QIR/C-API in the runtime for compilers
- the TOML schema for devices
- the Python entry points for frontends
- maybe others

I know they may not affect typical users, but they do affect integrators and aren't necessarily "internal" since they are meant for integration with other components or projects.

Reply:
Thanks @dime10 this is helpful :+1:

Reply:
I think it is up to us to make a decision here. I'm always a bit more conservative; we should err on something being internal _until_ we want to guarantee maintenance/not making breaking changes.

That is, the public API is what we declare to be the public API.

From the comments below, I'm inclined to say:

**Breaking change:**

- QuantumDevice interface (affects external plugin developers)
- TOML Schema

**Internal change:**
- Python entry points (since technically, the breaking change is in the PL API for compiler entrypoints, we are just conforming to it in this release)
- QIR/C-API in the runtime, based on Erick's comment below

Reply:
I agree on this assessment :)

Reply:
:100: